plugins/telegram-resources/src/components/Connect.svelte

```diff
@@ -13,151 +13,236 @@
 // limitations under the License.
 -->
 <script lang="ts">
-  import { getMetadata } from '@hcengineering/platform'
-  import { Button, EditBox, IconClose, Label } from '@hcengineering/ui'
+  import { getMetadata, IntlString } from '@hcengineering/platform'
+  import ui, { Button, EditBox, IconClose, Label } from '@hcengineering/ui'
   import { createEventDispatcher } from 'svelte'
   import presentation from '@hcengineering/presentation'
   import PinPad from './PinPad.svelte'
   import telegram from '../plugin'
   import { concatLink } from '@hcengineering/core'
+  import Error from '@hcengineering/ui/src/components/icons/Error.svelte'
 
-  const dispatch = createEventDispatcher()
+  type Integration = { status: 'authorized' | 'wantcode' | 'wantpassword', number: string } | 'Loading' | 'Missing'
+
+  const url = getMetadata(telegram.metadata.TelegramURL) ?? ''
 
-  let requested = false
-  let secondFactor = false
-  let connecting = false
   let phone: string = ''
   let code: string = ''
   let password: string = ''
-  let error: string | undefined = undefined
-  const url = getMetadata(telegram.metadata.TelegramURL) ?? ''
+  let error: string = ''
 
-  async function requestCode (): Promise<void> {
-    const res = await sendRequest('/signin', { phone })
-    if (res.next === 'code') {
-      requested = true
-    }
-
-    if (res.next === 'end') {
-      dispatch('close', { value: phone })
-    }
-  }
+  const dispatch = createEventDispatcher()
 
-  async function sendPassword (): Promise<void> {
-    const res = await sendRequest('/signin/pass', { phone, pass: password })
-    if (res.next === 'end') {
-      dispatch('close', { value: phone })
-    }
+  function close (): void {
+    dispatch('close')
   }
 
-  async function sendCode (): Promise<void> {
-    const res = await sendRequest('/signin/code', { phone, code })
-    if (res.next === 'pass') {
-      secondFactor = true
-    } else if (res.next === 'end') {
-      dispatch('close', { value: phone })
-    }
-  }
+  async function request (method: 'GET' | 'POST' | 'DELETE', path?: string, body?: any): Promise<any> {
```

Comment:
I would prefer to move api methods and types  in a separate file (api.ts for example) to make the component easier

Reply:
done